pytest_fixtures/component/discovery.py

```diff
@@ -0,0 +1,46 @@
+from fauxfactory import gen_string
+import pytest
+
+
+@pytest.fixture(scope='module')
+def module_discovery_hostgroup(module_org, module_location, module_target_sat):
+    host = module_target_sat.api.Host(organization=module_org, location=module_location)
+    host.create_missing()
+    return module_target_sat.api.HostGroup(
+        organization=[module_org],
+        location=[module_location],
+        medium=host.medium,
+        root_pass=gen_string('alpha'),
+        operatingsystem=host.operatingsystem,
+        ptable=host.ptable,
+        domain=host.domain,
+        architecture=host.architecture,
+    ).create()
+
+
+@pytest.fixture(scope='module')
+def discovery_org(module_org, module_target_sat):
+    # Update default discovered host organization
+    discovery_org = module_target_sat.api.Setting().search(
+        query={'search': 'name="discovery_organization"'}
+    )[0]
+    default_discovery_org = discovery_org.value
+    discovery_org.value = module_org.name
+    discovery_org.update(['value'])
+    yield module_org
+    discovery_org.value = default_discovery_org
+    discovery_org.update(['value'])
```

Comment:
```suggestion
    discovery_org = module_target_sat.update_setting('discovery_organization', module_org.name)
    yield
    module_target_sat.update_setting('discovery_organization', discovery_org)

Reply:
This will also require some changes @Gauravtalreja1 as previously it was a function scoped fixture in the UI module so leaving this as it is, and would come with a generic fixture with the UI scenario refactor.

Reply:
this is just a minor update to use helpers, it can be done in this PR and as we discussed refactor for this fixtures could be done in separate PR

Reply:
Done